packages/beacon-node/src/sync/sync.ts

```diff
@@ -57,14 +57,30 @@ export class BeaconSync implements IBeaconSync {
       this.rangeSync.on(RangeSyncEvent.completedChain, this.updateSyncState);
       this.network.events.on(NetworkEvent.peerConnected, this.addPeer);
       this.network.events.on(NetworkEvent.peerDisconnected, this.removePeer);
+      this.chain.clock.on(ClockEvent.epoch, this.onClockEpoch);
     } else {
       // test code, this is needed for Unknown block sync sim test
       this.unknownBlockSync.subscribeToNetwork();
       this.logger.debug("RangeSync disabled.");
-    }
 
-    // TODO: It's okay to start this on initial sync?
-    this.chain.clock.on(ClockEvent.epoch, this.onClockEpoch);
+      // In case node is started with `rangeSync` disabled and `unknownBlockSync` is enabled.
+      // If the epoch boundary happens right away the `onClockEpoch` will check for the `syncDiff` and if
+      // it's more than 2 epoch will disable the disabling the `unknownBlockSync` as well.
+      // This will result into node hanging on the head slot and not syncing any blocks.
+      // This was the scenario in the test case `Unknown block sync` in `packages/cli/test/sim/multi_fork.test.ts`
+      // So we are adding a particular delay to ensure that the `unknownBlockSync` is enabled.
+      const syncStartSlot = this.chain.clock.currentSlot;
+      // Having one epoch time for the node to connect to peers and start a syncing process
+      const epochCheckFotSyncSlot = syncStartSlot + SLOTS_PER_EPOCH;
```

Comment:
```suggestion
      const epochCheckForSyncSlot = syncStartSlot + SLOTS_PER_EPOCH;
```

Reply:
Will fix this typo in upcoming PR.

Reply:
Fixed by @tuyennhv already, see https://github.com/ChainSafe/lodestar/pull/6031/commits/1f36e8d5bc4696b369417d15627c461fd951d0e1